src/main/java/org/jenkins_ci/plugins/run_condition/core/StringsMatchCondition.java

```diff
@@ -32,17 +32,26 @@
 import org.jenkinsci.plugins.tokenmacro.TokenMacro;
 import org.kohsuke.stapler.DataBoundConstructor;
 
+import java.util.Map;
+import java.util.logging.Logger;
+
 public class StringsMatchCondition extends AlwaysPrebuildRunCondition {
 
     final String arg1;
     final String arg2;
     final boolean ignoreCase;
+    final boolean shellVariables;
```

Comment:
Don't make it `final` so that it can be set from the `DataBoundSetter`
```suggestion
    boolean shellVariables;
```

Reply:
> Thanks for the pull request.
> 
> I'm not sure of the use case that this is addressing. Can you add a documentation proposal for text in the README? That is a good place to describe how it will be used and how it will help users.
> 
> If this is fixing a specific issue report, please include a link to the issue report in the pull request description.
> 
> Please add automated tests that confirm the modified code is behaving as expected. An enhancement without automated tests is much more likely to be harmed by later changes than an enhancement that includes automated tests.

![new](https://user-images.githubusercontent.com/60972989/190361838-b46b0ba3-e750-4430-9129-6890aca63866.jpg)
Sir, In the current version, there are no test cases is present as of now related to **StringMatchCondition.java**